terraform/file-hosting/vcl/files.vcl

```diff
@@ -139,6 +139,30 @@ sub vcl_fetch {
         }
     }
 
+    # Check if we are serving a .metadata file, which are stored uncompressed
+    if (req.url ~ "^/packages/[a-f0-9]{2}/[a-f0-9]{2}/[a-f0-9]{60}/(.*).metadata$" ) {
+        # Always set a Vary header, even if we don't end up compressing
+        # the object, because the uncompressed version should only be
+        # used when the request does NOT request the compressed one.
+        if (!beresp.http.Vary ~ "Accept-Encoding") {
+            if (beresp.http.Vary) {
+                set beresp.http.Vary = beresp.http.Vary ", Accept-Encoding";
+            } else {
+                set beresp.http.Vary = "Accept-Encoding";
+            }
+        }
+
+        # Perform compression if the client claims to understand it
+        if (req.http.Accept-Encoding == "gzip") {
+            set beresp.gzip = true;
+        } else if (req.http.Accept-Encoding == "br") {
```

Comment:
Hi! :wave:

Is there a particular reason to prefer gzip over brotli? Especially for perdominantly English vocabulary, brotli outperforms gzip both in io times and compression ratio.

Here's a quick benchmark I did recently: https://github.com/illagrenan/django-brotli/pull/117

Reply:
What happens if the client accepts multiple encodings?

Like requests and [aiohttp](https://github.com/aio-libs/aiohttp/blob/f5df67aa5bb5a52e7185659673e98301b0a59855/aiohttp/client_reqrep.py#L88) and probably most others do?

Reply:
I don't think there's any reason to prefer gzip over brotli, though I suspect the answer is that it doesn't matter regardless because 99% of the traffic will be gzip anyways because the primary consumer will almost certainly be pip, which uses requests which, afaik, doesn't support brotli by default (requires extra libraries, and likely would get disabled in pip anyways because C libraries cant be upgraded on Windows if pip imports them).

Reply:
At the point in the request that this happens, the [``Accept-Encoding`` header is normalized to only be a single encoding](https://developer.fastly.com/reference/http/http-headers/Accept-Encoding/).